synapse/rest/client/v1/admin.py

```diff
@@ -66,6 +69,25 @@ def on_GET(self, request, user_id):
         defer.returnValue((200, ret))
 
 
+class VersionServlet(ClientV1RestServlet):
+    PATTERNS = client_path_patterns("/admin/server_version")
+
+    @defer.inlineCallbacks
+    def on_GET(self, request):
+        requester = yield self.auth.get_user_by_req(request)
+        is_admin = yield self.auth.is_server_admin(requester.user)
+
+        if not is_admin:
+            raise AuthError(403, "You are not a server admin")
+
+        ret = {
+            'server_version': get_version_string(synapse),
+            'python_version': platform.python_version(),
+        }
```

Comment:
Then there's what should be returned. @hawkowl suggested to also return the Python version (IIRC) because we can, and because this is often a pain point for users.

Reply:
> this is often a pain point for users.

i.e. that troubleshooting is harder without this information, so we may as well return it